lightly/api/api_workflow_sampling.py

```diff
@@ -37,7 +40,12 @@ def sampling(self, sampler_config: SamplerConfig, al_scores: Dict[str, List[int]
 
         # upload the active learning scores to the api
         if al_scores is not None:
-            raise NotImplementedError  # TODO: fill out in later branches
+            if preselected_tag_id is None:
+                raise ValueError
+            for score_type, score_values in al_scores.items():
```

Comment:
I'd suggest to make a check in the pip package whether the number of scores coincides with the size of the tag they're being uploaded to. @MalteEbner @IgorSusmelj thoughts on that?

Reply:
I did that in the ActiveLearningAgent.query()

Reply:
Didn't see that, thanks for the hint